.github/workflows/test-cygwin.yml

```
@@ -0,0 +1,113 @@
+name: Test Cygwin
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  build:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-minor-version: [7, 8, 9]
+        architecture: ["x86", "x86_64"]
+
+    timeout-minutes: 40
+
+    name: Python 3.${{ matrix.python-minor-version }} ${{ matrix.architecture }}
+
+    steps:
+      - name: Fix line endings
+        run: |
+          git config --global core.autocrlf input
+
+      - name: Checkout Pillow
+        uses: actions/checkout@v3
+
+      - name: Install Cygwin
+        uses: cygwin/cygwin-install-action@v2
+        with:
+          platform: ${{ matrix.architecture }}
+          packages: >
+            ImageMagick jpeg python3${{ matrix.python-minor-version }}-cffi
+            python3${{ matrix.python-minor-version }}-devel
+            python3${{ matrix.python-minor-version }}-numpy
+            python3${{ matrix.python-minor-version }}-cython liblapack-devel gcc-g++
+            python3${{ matrix.python-minor-version }}-sip
+            python3${{ matrix.python-minor-version }}-tkinter ghostscript
+            libfreetype-devel libimagequant-devel libjpeg-devel liblcms2-devel
+            libopenjp2-devel libraqm-devel libtiff-devel libwebp-devel libxcb-devel
+            libxcb-xinerama0 netpbm perl qt5-devel-tools xorg-server-extra zlib-devel
+            subversion make
+
+      - name: Add Lapack to PATH
+        uses: egor-tensin/cleanup-path@v1
+        with:
+          dirs: 'C:\cygwin\bin;C:\cygwin\lib\lapack'
+          
+      - name: pip cache
+        uses: actions/cache@v3
+        with:
+          path: 'C:\cygwin\home\runneradmin\.cache\pip'
+          key: ${{ runner.os }}-cygwin-${{ matrix.architecture }}-pip3.${{ matrix.python-minor-version }}-${{ hashFiles('.ci/install.sh') }}
+          restore-keys: |
+            ${{ runner.os }}-cygwin-${{ matrix.architecture }}-pip3.${{ matrix.python-minor-version }}-
+
+      - name: Build system information
+        run: |
+          bash.exe -c "python3 .github/workflows/system-info.py"
+
+      - name: Install dependencies
+        run: |
+          bash.exe .ci/install.sh
```

Let's silence these warnings:

```
WARNING: You are using pip version 21.3.1; however, version 22.0.4 is available.
You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
```

```suggestion
          bash.exe -c "python3.${{ matrix.python-minor-version }} -m pip install -U pip"
          bash.exe .ci/install.sh
```